scripts/build_podspec.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the Swift Logging API open source project
+##
+## Copyright (c) 2019 Apple Inc. and the Swift Logging API project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of Swift Logging API project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftNIO open source project
+##
+## Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+set -eu
+
+function usage() {
+  echo "$0 [-u] version"
+  echo
+  echo "OPTIONS:"
+  echo "  -u: Additionally upload the podspec"
+}
+
+upload=false
+while getopts ":u" opt; do
+  case $opt in
+    u)
+      upload=true
+      ;;
+    \?)
+      usage
+      exit 1
+      ;;
+  esac
+done
+shift "$((OPTIND-1))"
+
+if [[ $# -eq 0 ]]; then
+  echo "Must provide target version"
+  exit 1
+fi
+
+version=$1
+NAME="SwiftLogAPI"
+
+here="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+tmpdir=$(mktemp -d /tmp/.build_podspecsXXXXXX)
+echo "Building podspec in $tmpdir"
+
+cat > "${tmpdir}/${NAME}.podspec" <<- EOF
+Pod::Spec.new do |s|
+  s.name = '$NAME'
+  s.version = '$version'
+  s.license = { :type => 'Apache 2.0', :file => 'LICENSE.txt' }
+  s.summary = 'A Logging API for Swift.'
+  s.homepage = 'https://github.com/apple/swift-log'
+  s.author = 'Apple Inc.'
+  s.source = { :git => 'https://github.com/apple/swift-log.git', :tag => s.version.to_s }
+  s.documentation_url = 'https://github.com/apple/swift-log'
+  s.module_name = 'Logging'
+
+  s.swift_version = '5.0'
+  s.cocoapods_version = '>=1.6.0'
+  s.ios.deployment_target = '12.0'
+  s.osx.deployment_target = '10.14'
+  s.tvos.deployment_target = '12.0'
+
+  s.source_files = 'Sources/Logging/**/*.swift'
+end
+EOF
+
+if $upload; then
```

Comment:
worth calling `pod spec lint "${tmpdir}/${NAME}.podspec"` here?

Reply:
@tomerd it does the linting itself automatically

Reply:
```
[!] No podspec found in directory `.`
Usage:

    $ pod trunk push [PATH]

      Publish the podspec at `PATH` to make it available to all users of the ‘master’
      spec-repo. If `PATH` is not provided, defaults to the current directory.

      Before pushing the podspec to cocoapods.org, this will perform a local lint of
      the podspec, including a build of the library. However, it remains *your*
      responsibility to ensure that the published podspec will actually work for your
      users. Thus it is recommended that you *first* try to use the podspec to
      integrate the library into your demo and/or real application.

      If this is the first time you publish a spec for this pod, you will
      automatically be registered as the ‘owner’ of this pod. (Note that ‘owner’ in
      this case implies a person that is allowed to publish new versions and add other
      ‘owners’, not necessarily the library author.)
```

Reply:
right, but when you do local dev, you dont set `upload` (push) so this help validating imo. no big deal tho if you think it does not add value

Reply:
@tomerd I'll actually add it if you don't set `upload`, right?

Reply:
that's a good point because then you can validate before actually uploading ;)

Reply:
yes something like https://github.com/tomerd/swift-metrics/blob/feature/podspec/scripts/build_podspec.sh#L87

Reply:
done